docs/installation-guide.md

```diff
@@ -0,0 +1,67 @@
+---
+title: "Installation Guide"
+weight: 20
+---
+
+You can use the Generator library to generate whatever you want in your event-driven architecture apps. Find your preferred method below:
+- [Generator CLI](#generator-cli)
+- [Use the Generator Library in Node.js Apps](#2-use-the-generator-library-in-your-nodejs-apps)
+  
+## Prerequisites
+Before you install and use the AsyncAPI CLI and the generator library, ensure you meet the prerequisites below, then [install the CLI](#installation).
+1. Node.js v12.16 and higher
+2. Npm v6.13.7 and higher
+   
+To verify the versions of Node and Npm you have, run the following command on your terminal:
+```
+npm -v
+```
+```
+node -v
+```
+
+If you don't have either Node or Npm installed, use the [official node.js installer](https://nodejs.org/en/download/).
+
+If you have the correct versions installed, proceed to the CLI installation guide below. Otherwise, upgrading the Npm or Node version is lower than the recommended versions specified above.
+
+## Generator CLI
+The CLI tool allows you to generate message-based API boilerplate code, documentation, or anything else you need as long as you specify it in your [template](template.md). To use the Generator via the CLI, you will need to install the AsyncApi Generator CLI tool in your local development environment.
+
+### Installation
+
+The recommended method to interact with the generator tool is its CLI. To use the CLI, install it  using the following `npm` command:
+```
+npm install -g @asyncapi/generator
+```
+
+To install a specific version of the generator tool, pass the version during installation:
+```
+npm install -g @asyncapi/generator@{version}
+```
+> :bulb: **Remember:** 
+> Each [community developed template](https://github.com/search?q=topic%3Aasyncapi+topic%3Agenerator+topic%3Atemplate) is dependent on a certain version of the generator for it to work correctly. Before you install the generator CLI, check the templates' `package.json` for the version of the Generator CLI your template is compatible with. To learn more about the generator version and how it correlates to the template you use check the [versioning docs](versioning.md).
+
```

Comment:
Interesting ! I think this is very long and verbose, without telling us much at all. However, I DO think we should include this in this installation section because documenting diff OS installation is a best practice. Here is a standard example from Gastby that shows how this can look like... 

<img width="400" alt="Screen Shot 2022-10-19 at 2 07 36 PM" src="https://user-images.githubusercontent.com/19964402/196804641-ed353956-ea1d-414f-a15b-00b58407a12e.png">

## My feedback suggestion
Florence, I'd prob shorten this and improve it by making a tiny section per top two OS minimum:

```
#### MacOS 
You can install in MacOS by using brew: `brew install asyncapi`. 

#### Linux 
You can install in Linux by using `dpkg`, a package manager for debian:
1. 'curl -OL https://github.com/asyncapi/cli/releases/latest/download/asyncapi.deb`
2. `sudo dpkg -i asyncapi.deb`

#### Other Operating Systems
For further installation instructions for different operating systems, read the [`AsyncAPI CLI` readME file](https://github.com/asyncapi/cli#installation)
```

Reply:
ok, and overtime we can always ask volunteers to update doc as first time contribution, of there are more methods available 👍🏼